### PR TITLE
Core/Scripts: Fixed Naxx Anub Scarab Summoning Spell

### PR DESCRIPTION
**Pull request by HelloKitty**
**Changes proposed:**

-  Added a single SpellScript for Spells [29105](http://www.wowhead.com/spell=29105/summon-corpse-scarabs-5) and [28864](http://www.wowhead.com/spell=28864/summon-corpse-scarabs-10)
-  Added SQL to spell_script_names for the spell script.

**Target branch(es):**

- [x] 3.3.5

**Issues addressed:**

Fixes the issue of a broken default spell implementation. The spell summoned scarabs in a 100y radius which wasn't what the fight was suppose to do.

**Tests performed:**

Tested it in-game. The scarabs are now summoned properly.

**Known issues and TODO list:**

NONE

**Commit Message**

The scarab summoning spell 29105 and 28864 were broken and were spawning
scarabs in a 100y radius which broke the fight's mechanic.

### CODE REVIEW

File: src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp [hunk 1]
```diff
@@ -256,9 +257,51 @@ class at_anubrekhan_entrance : public AreaTriggerScript
...
+    spell_summon_corpse_scarabs() : SpellScriptLoader("spell_summon_corpse_scarabs") { }
+
+    class spell_summon_corpse_scarabs_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_summon_corpse_scarabs_SpellScript);
+
+        void OnEffectHitHandler(SpellEffIndex effIndex)
+        {
+            //disables the default spawning of scarabs in a 100y radius.
+            this->PreventHitDefaultEffect(effIndex);
```
Comment (Keader) [inline src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp]: remove ```this->```

File: src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp [hunk 2]
```diff
@@ -256,9 +257,51 @@ class at_anubrekhan_entrance : public AreaTriggerScript
...
+
+            if (Unit* targetUnit = GetCaster()) //target should be the same as the caster
+            {
+                if (Unit* originalCaster = this->GetOriginalCaster()) //original caster is used to determine who should control the summoning.
```
Comment (Keader) [inline src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp]: remove ```this->```

**Comment by joschiwald** [comment]
i would suggest to use a SpellInfoCorrection

**Comment by HelloKitty** [comment]
@joschiwald What is that?

**Comment by joschiwald** [comment]
SpellMgr::LoadSpellInfoCorrections

**Comment by Krudor** [comment]
This is a hack fix and does not mimic blizzlike behavior.
They scarabs are not spawned randomly inside a radius ,they are spawned exactly ontop of eachother.
You ave overriding the blizzlike behavior, and this is not the place to fix it.

**Comment by HelloKitty** [comment]
This spell is nonfunctional and it spawns the scarabs in. 100y radius all over the place including in the walls and usually far away from the encounter.

That is not Blizzlike behavior I think and I know this because not only do videos show that the scarabs don't spawn inside walls but when I played back in 2009 they didn't do that either.

On top of this I think you're also wrong in that they spawn all on top of eachother. The reference video I found did not show that behavior. I'd be happy to look further into it but either way the spell as-is in TC is not functional and must be fixed in some way.

**Comment by HelloKitty** [comment]
@Krudor So either way, whether they spawn in a single point or in a small radius, the spell is currently 100% broken on Trinitycore and needs to be fixed. It's just a simple change in this PR to make them spawn in a single point.

**Comment by Kittnz** [comment]
They actaully do spawn ontop of eachother.

**Comment by Treeston** [comment]
Yeah, that's broken on current TC. They should all spawn in a stack, but don't.

I believe it's a bug with all spells using the same targeting behavior.

**Comment by jackpoz** [comment]
> i would suggest to use a SpellInfoCorrection

@joschiwald could you post the correction you would apply ?

**Comment by joschiwald** [comment]
```diff
diff --git a/src/server/game/Spells/SpellMgr.cpp b/src/server/game/Spells/SpellMgr.cpp
index e49ac050c5..807ec4c1f5 100644
--- a/src/server/game/Spells/SpellMgr.cpp
+++ b/src/server/game/Spells/SpellMgr.cpp
@@ -2836,6 +2836,12 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->SpellFamilyFlags[2] = 0x80000000;
     });
 
+    // Summon Corpse Scarabs
+    ApplySpellFix({ 28864, 29105 }, [](SpellInfo* spellInfo)
+    {
+        const_cast<SpellEffectInfo*>(spellInfo->GetEffect(EFFECT_0))->RadiusEntry = sSpellRadiusStore.LookupEntry(EFFECT_RADIUS_10_YARDS);
+    });
+
     //
     // VIOLET HOLD SPELLS
     //
```
(bases on master branch)

**Comment by Krudor** [comment]
EFFECT_RADIUS_0_YARDS

**Comment by tkrokli** [comment]
So the `Effect Radius: 100 yards` on wowhead is not related for those 2 spells?

**Comment by ccrs** [comment]
never use wowhead as reference

**Comment by jackpoz** [comment]
@joschiwald is your patch enough to fix the issue, making this PR obsolete ?

**Comment by jackpoz** [comment]
handled by https://github.com/TrinityCore/TrinityCore/commit/72e97d330bd4d696255eee7046a27a9f8b3d6ae1 with joschinwald patch

**Pull request closed without merging**